### DISCUSSION TITLE
Function for Initial_condition

### DISCUSSION THREAD

**Discussion by ananthanarasimhanj** [idaholab/moose#27417] [Q&A General]
I have a variable "e" as shown in the below code.  I want to expression (or function) in the place of constant value given to "initial_condition". what should I change? 

```
[Variables]
  [e]
    family = SCALAR
    order = FIRST
    initial_condition = 13.81551056 
    #scaling = 1e-3
  []
[]

[ScalarKernels]

  [de_dt]
    type = ODETimeDerivativeLog
    variable = e
  []

[]
```

**Comment by GiudGiud** [comment]
Hello

You use the Scalar----ICs, in the [ICs] block to set an initial condition to ScalarVariables.
Here's a link to the one that can use a function
https://mooseframework.inl.gov/source/ics/FunctionScalarIC.html

Guillaume

**Reply by ananthanarasimhanj** [reply]
Sure. Is it also allowed to do this through AuxVariables/AuxScalarKernels?

**Reply by GiudGiud** [reply]
That s ok too

You can execute them on InITIAL it is also the same as an initial condition

**Reply by ananthanarasimhanj** [reply]
Thanks... below one looks OK? 

```
[AuxScalarKernels]
  [H2]
    type = ParsedAuxScalar
    variable = H2
    args = 'Nd'
    function = 'log(Nd)'
    execute_on = 'INITIAL'
  []
```

**Reply by ananthanarasimhanj** [reply]
There is also another question for me... this variable I want to initialize using an expression is a chemical species, and it has to have time derivative kernel. Is it possible to use an expression for the nonlinear variable instead of using as aux? for example below code, where initial condition is a constant has to be replaced with an expression. How should we do that? 


```
[Variables]
  [H2]
    family = SCALAR
    order = FIRST
    initial_condition = 58
  []
 []

[ScalarKernels]

  [H2]
    type = ODETimeDerivativeLog
    variable = H2
  []
 []
```

**Reply by GiudGiud** [reply]
Hello

For more complex intiializations you use the ICs block instead of specify it directly in the Variable

There are many ICs objects you can use.

Auxkernel on initial are also still an option imo

Guillaume

**Reply by ananthanarasimhanj** [reply]
But, variable in AuxKernel, its value will be held constant throughout the simulation and cannot be subjected to a time derivative as this is considered to be a chemical species in this problem statement. Isnt it?

**Reply by GiudGiud** [reply]
They can be modified by auxkernels on each time step
I dont know actually if the time derivative is not computed. It would be worth checking
